finagle-core/src/test/scala/com/twitter/finagle/service/StatsFilterTest.scala

```diff
@@ -5,22 +5,37 @@ import com.twitter.finagle.stats.{CategorizingExceptionStatsHandler, ExceptionSt
 import com.twitter.finagle._
 import com.twitter.util._
 import java.util.concurrent.TimeUnit
+
 import org.junit.runner.RunWith
 import org.scalatest.FunSuite
 import org.scalatest.junit.JUnitRunner
 
+import scala.util.control.ControlThrowable
+
 @RunWith(classOf[JUnitRunner])
 class StatsFilterTest extends FunSuite {
   val BasicExceptions = new CategorizingExceptionStatsHandler(_ => None, _ => None, rollup = false)
 
+  sealed trait ServiceBehaviour
```

Comment:
instead of adding all of this extra machinery to the `getService` method, we should just inline a new service.